cloud/language_modeling/benchmark.py

```diff
@@ -0,0 +1,306 @@
+
+        for i in range(self.miniBatchSize):
+            inputs[i, :] = self.nextSample()
+
+        return inputs[:, 0:self.wordsPerSample], inputs[:, 1:]
+
+    def nextSample(self):
+        sample = self.samples[self.index]
+        self.advance()
+        return sample
+
+    def advance(self):
+        self.index += 1
+
+def downloadValidationDataset(arguments):
+    import os
+    import urllib.request
+    filename = "validation-dataset.tar.gz"
+    if not os.path.exists(filename):
+        print("Downloading dataset from " + arguments["validation_dataset_url"])
+        urllib.request.urlretrieve(arguments["validation_dataset_url"], filename)
+
+    print("extracting validation dataset " + filename)
+    extract(filename, 'validation-dataset')
+
+def loadVocab(arguments):
+    import os
+    path = arguments["vocab_path"]
+
+    if not os.path.exists(path):
+        raise RuntimeError("Could not locate validation data at: " + path)
+
+    vocab = {}
+
+    with open(path, "r") as vocabFile:
+        for line in vocabFile:
+            vocab[line.strip()] = len(vocab)
+
+    vocab['UNKNOWN'] = len(vocab)
+
+    return vocab
+
+def loadValidationDataset(arguments):
+    import os
+    vocab = loadVocab(arguments)
+
+    path = "validation-dataset/training-monolingual/news-commentary-v6.en"
+
+    if not os.path.exists(path):
+        raise RuntimeError("Could not locate validation data at: " + path)
+
+    dataset = Dataset(vocab, int(arguments["words_per_sample"]), int(arguments["maximum_samples"]),
+        int(arguments["mini_batch_size"]))
+
+    with open(path, "r") as datasetFile:
+        for line in datasetFile:
+            for word in line.split():
+                dataset.addWord(word)
+
+                if dataset.isFull():
+                    break
+
+            if dataset.isFull():
+                break
+
+    return dataset
+
+def getValidationDataset(arguments):
+    if not compareChecksum("validation-dataset", arguments["validation_checksum"]):
+        downloadValidationDataset(arguments)
+
+    if not compareChecksum("validation-dataset", arguments["validation_checksum"]):
+        raise RuntimeError("Checksum mismatch on validation dataset.")
+
+    return loadValidationDataset(arguments)
+
+def getModel(arguments):
+    if not compareChecksum("model-checkpoint", arguments["model_checksum"]):
+        downloadModel(arguments)
+
+        if not compareChecksum("model-checkpoint", arguments["model_checksum"]):
+            raise RuntimeError("Checksum mismatch on model.")
+    else:
+        print("Checksum for model passed, loading model...")
+
+    return loadModel()
+
+def main():
+    from argparse import ArgumentParser
+
+    parser = ArgumentParser(description="MLPerf Inference Language Modeling Benchmark.")
+
+    parser.add_argument("-m", "--model-url",
+        default = "https://zenodo.org/record/1492892/files/big-lstm.tar.gz?download=1",
```

Comment:
Can I trouble you to please update this to the new Zenodo link?

The new Zenodo link being where you upload the model into the https://zenodo.org/communities/mlperf/ repository. Thanks!